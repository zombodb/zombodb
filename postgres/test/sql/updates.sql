set datestyle to 'iso, mdy';

SELECT assert('# of posts', 165240, count(*)) FROM so_posts;

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert('# of posts', 165240, count(*)) FROM so_posts;
SELECT assert('estimated count', 1000 + (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));
COMMIT;

SELECT assert('estimated count', (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert('# of posts', 165240, count(*)) FROM so_posts;
SELECT assert('estimated count', 1000 + (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));
ABORT;

SELECT assert('estimated count', (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));

