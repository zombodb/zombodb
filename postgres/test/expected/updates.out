set datestyle to 'iso, mdy';
SELECT assert('# of posts', 165240, count(*)) FROM so_posts;
 assert 
--------
 165240
(1 row)

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert('# of posts', 165240, count(*)) FROM so_posts;
 assert 
--------
 165240
(1 row)

SELECT assert('estimated count', 1000 + (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));
 assert 
--------
 166240
(1 row)

COMMIT;
SELECT assert('estimated count', (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));
 assert 
--------
 165240
(1 row)

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert('# of posts', 165240, count(*)) FROM so_posts;
 assert 
--------
 165240
(1 row)

SELECT assert('estimated count', 1000 + (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));
 assert 
--------
 166240
(1 row)

ABORT;
SELECT assert('estimated count', (SELECT count(*) FROM so_posts), zdb_estimate_count('so_posts', ''));
 assert 
--------
 165240
(1 row)

